### PR TITLE
Pnp continue

### PR DESCRIPTION
**Pull request by igor-dv**
Based on #8

### CODE REVIEW

**Comment by igor-dv** [comment]
@ndelangen @shilman ☝️ it's ready.

We will need to release `preset-typescript` as well.

File: packages/preset-typescript/index.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
-function webpack(webpackConfig = {}, options = {}) {
+async function webpack(webpackConfig = {}, options = {}) {
   const { module = {}, resolve = {} } = webpackConfig;
-  const { tsLoaderOptions, tsDocgenLoaderOptions, include } = options;
+  const { tsLoaderOptions, tsDocgenLoaderOptions, include, presets } = options;
+
+  const loaderOptions = await presets.apply('tsLoaderOptions', tsLoaderOptions, { presets });
```
Comment (shilman) [inline packages/preset-typescript/index.js]: I don't fully understand this line. Where is presets defined? Is there an example usage somewhere I can look at?

**Comment by drphelps** [comment]
@ndelangen @shilman @igor-dv What is the status of this PR please?

File: packages/preset-typescript/index.js (hunk 1 above)
Comment (ndelangen) [reply packages/preset-typescript/index.js]: presets is destructured from options, this is how preset work.

**Comment by ndelangen** [comment]
@igor-dv do you plan on finishing this? @drphelps would you be able to help maybe?

**Comment by gaetanmaisse** [comment]
This PR is maybe not useful anymore as we are working on supporting Yarn 2 (and so PnP) in SB directly?
 - Issue: https://github.com/storybookjs/storybook/issues/9527
 - Main PR: https://github.com/storybookjs/storybook/pull/9667

**Comment by gaetanmaisse** [comment]
Closing as PnP support has been handled in other PRs.

**Pull request closed without merging**